### PR TITLE
bpf: Only create veth pair if it does not already exist.

### LINKED ISSUE #4198: L7 proxy stops receiving requests after cilium config ConntrackLocal=true
Replication steps:
```bash
set -x

sudo service cilium stop
sudo docker rm -f `docker ps -aq`
sudo cilium cleanup -f
sudo service cilium restart
until sudo cilium status; do sleep 1s; done
sudo cilium service delete --all
docker run -d --name app2 --net cilium -l id.app2 tgraf/netperf 
docker run -d --name app3 --net cilium -l id.app3 tgraf/netperf 
docker run -d --name httpd1 --net cilium -l id.httpd1 cilium/demo-httpd 
docker run -d --name httpd2 --net cilium -l id.httpd2 cilium/demo-httpd 
docker run -d --name httpd3 --net cilium -l id.httpd3 cilium/demo-httpd 
docker run -d --name app1 --net cilium -l id.app1 tgraf/netperf 
#sudo cilium monitor -v | ts '[%Y-%m-%d %H:%M:%S]'
#cilium endpoint list -o jsonpath='{range [*]}{@.status.external-identifiers.container-name}{"="}{@.status.state},{@.status.identity.id}{"\n"}{end}'

sleep 10s
#sudo cilium config ConntrackLocal=true
sudo cilium endpoint list

HTTPD1_IPV4=$(sudo cilium endpoint list | grep httpd1 | awk '{print $7}')
HTTPD2_IPV6=$(sudo cilium endpoint list | grep httpd2 | awk '{print $6}')
HTTPD2_IPV4=$(sudo cilium endpoint list | grep httpd2 | awk '{print $7}')

sudo cilium service update --frontend '2.2.2.100:80' --backends "${HTTPD1_IPV4}:80" --id '100' --rev
sudo cilium service update --frontend '[f00d::1:1]:80' --backends "[${HTTPD2_IPV6}]:80" --id '101' --rev
sudo cilium service update --frontend '2.2.2.101:80' --backends "${HTTPD2_IPV4}:80" --id '102' --rev
sudo cilium policy get -o json | jq '.revision'
sudo cilium policy validate ./test/runtime/manifests/Policies-l7-simple.json
sudo cilium policy import ./test/runtime/manifests/Policies-l7-simple.json
sudo cilium policy get -o json | jq '.revision'
sudo cilium policy wait 7
sudo cilium policy get -o json | jq '.revision'
sudo cilium endpoint list
echo "should pass"
docker exec -i  app1 curl -s --fail --connect-timeout 5 --max-time 5 http://2.2.2.100:80/public
echo "should fail"
docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://2.2.2.100:80/public
echo "should pass"
docker exec -i  app2 curl -s --fail --connect-timeout 5 --max-time 5 http://[f00d::1:1]:80/public
echo "triggering the bug"
sudo cilium config ConntrackLocal=true
sleep 10s
echo "should pass"
docker exec -i  app1 curl -s --fail --connect-timeout 5 --max-time 5 http://2.2.2.100:80/public
echo "should fail"
docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://2.2.2.100:80/public
echo "should pass"
docker exec -i  app2 curl -s --fail --connect-timeout 5 --max-time 5 http://[f00d::1:1]:80/public
```

after "triggering the bug" step / setting the `ConntrackLocal=true`, all requests seem to be dropped.

If `sudo cilium config ConntrackLocal=true` is done before the first 3 curls. the L7-proxy, allows/denies as expected

### PR DESCRIPTION
**Pull request by jrajahalme**
Deleting and re-creating a veth pair will pick up new MAC addresses
and interface indices, avoid that by figuring out if the veth pair
already exists.

This fixes a frequest CI issue, but also hides underlying problems
relating to endpoint bpf regeneration on daemon config changes or
restarts. New GH issues are opened to keep track of those.

Fixes: #4198
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4214)

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

File: bpf/init.sh [hunk 1]
```diff
@@ -86,8 +86,12 @@ function setup_veth_pair()
 	local -r NAME1=$1
 	local -r NAME2=$2
 
-	ip link del $NAME1 2> /dev/null || true
-	ip link add $NAME1 type veth peer name $NAME2
+	# Only recreate the veth pair if it does not exist already.
+	# This avoids propblems with changing MAC addresses.
```
Comment (rlenglet) [inline bpf/init.sh]: propblems -> problems

**Comment by jrajahalme** [comment]
test-me-please

File: bpf/init.sh (hunk 1 above)
Comment (jrajahalme) [reply bpf/init.sh]: Fixed, thanks!

Comment (jrajahalme) [reply bpf/init.sh]: First CI build succeeded :-)

**Review by joestringer — APPROVED** [review]
We reviewed this offline together to validate that `set -e` won't cause weird behaviour inside the if statement. Looks correct, a green CI build should provide reasonable assurance that it doesn't cause regressions.

**Comment by jrajahalme** [comment]
Forgot to update bindata after the typo fix => CI failed. Trying again.

**Comment by jrajahalme** [comment]
test-me-please

**Comment by aanm** [comment]
This is an interesting one. Great catch! cc @eloycoto you will be happy with this one

**Review by borkmann — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
@jrajahalme needs rebase

**Comment by jrajahalme** [comment]
Rebased.

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
Another rebase (after a successful CI run)

**Comment by jrajahalme** [comment]
test-me-please

**Pull request merged**